### ISSUE TITLE
alertmanage sends recovery message bug

### ISSUE THREAD

**Issue by guleng** [coreos/kube-prometheus#225]
**Alertmanager sends recovery messages multiple times**

**Environment**

* Prometheus Operator version:

    `alertmanager:v0.16.1  ,  prometheus:v2.7.2`

* Kubernetes version information:

    `kubectl version:v1.14.1`
    

```
example：
Alarm name：node-down
     Warning details  : compile03.yun.com node is down. 
     Alarm level:: critical
     Trigger time: 2019-09-13 10:53:34
      ---
     Warning details  : gpu412.yun.com node is down. 
     Alarm level:: critical
     Trigger time: 2019-09-14 09:14:34
      ---
     Restore details : gpu802.yun.com node is down. 
     Recovery level: critical
     Trigger time: 2019-09-14 02:35:34
     recovery time: 2019-09-14 09:01:34
```
```
Alarm name：node-down
     Warning details  : compile03.yun.com node is down. 
     Alarm level:: critical
     Trigger time: 2019-09-13 10:53:34
      ---
     Restore details  : gpu412.yun.com node is down. 
     Recovery level   : critical
     Trigger time:  2019-09-14 09:14:34
     recovery time:2019-09-14 09:18:34 
```
```
Alarm name：node-down
     Warning details  : compile03.yun.com node is down. 
     Alarm level:: critical
     Trigger time: 2019-09-13 10:53:34
      ---
     Warning details  : gpu502.yun.com node is down. 
     Alarm level:: critical
     Trigger time: 2019-09-14 09:59:04 
```
```
Here compile03.yun.com takes the message with it at the same time when it is restored or triggered, and it feels wrong.
```

**Comment by brancz** [comment]
cc @simonpasquier

**Comment by simonpasquier** [comment]
I don't understand what is the issue. Can you rephrase please?
Can you share your rules + AlertManager configuration?

**Comment by guleng** [comment]
It means that the alarm is repeated many times before it returns to normal.
If there is another new alarm before the old one has returned to normal, the new alarm will be sent out together with the old one.

So before my repeat_interval: 1d   time arrived, the old alarm was repeated.

```
My configuration：
  group_wait: 30s
  group_interval: 2m
  repeat_interval: 1d
```

**Comment by simonpasquier** [comment]
Hmm, I'm not quite getting the issue. The notification message sent by Alertmanager includes the firing and resolved alerts. If the sequence is 
1. alert firing for node A
1. alert firing for node B
1. alert resolved for node A

Then the respective notifications would be:
1. "alert firing for node A"
1. "alert firing for node A" + "alert firing for node B"
1. "alert resolved for node B" + "alert firing for node B"

**Comment by guleng** [comment]
Yes, right now, there are repeated A's.

**Comment by simonpasquier** [comment]
This is working as expected then. Alertmanager evaluate the alerts group at every `group_interval` and if something has changed (like an existing alert being resolved or a new alert firing), it will notify all the alerts, not the new alerts only.
